gax-java/gax/src/test/java/com/google/api/gax/core/GaxPropertiesTest.java

```diff
@@ -57,10 +58,85 @@ public void testGaxVersion() {
     }
   }
 
+  private static String originalJavaVersion = System.getProperty("java.version");
+  private static String originalJavaVendor = System.getProperty("java.vendor");
+  private static String originalJavaVendorVersion = System.getProperty("java.vendor.version");
+
+  @After
+  public void cleanup() {
+    if (Strings.isNullOrEmpty(originalJavaVersion)) {
+      System.clearProperty("java.version");
+    } else {
+      System.setProperty("java.version", originalJavaVersion);
+    }
+
+    if (Strings.isNullOrEmpty(originalJavaVendor)) {
+      System.clearProperty("java.vendor");
+    } else {
+      System.setProperty("java.vendor", originalJavaVendor);
+    }
+
+    if (Strings.isNullOrEmpty(originalJavaVendorVersion)) {
+      System.clearProperty("java.vendor.version");
+    } else {
+      System.setProperty("java.vendor.version", originalJavaVendorVersion);
+    }
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfo_spaces() {
```

Comment:
Can you rename it back to the Java vendor name? For those that represent real vendor value, the name should have the vendor name.

Reply:
Is the purpose of this unit test to test real vendor name or test the functionality of the method?

Reply:
@ddixit14 The test name should answer Blake'a question.

Reply:
I have tried to capture both.

Reply:
I really think that unit tests should only care about the functionality/behavior of the method and [test names](https://g3doc.corp.google.com/eng/doc/devguide/testing/unit/best-practices.md?cl=head&polyglot=java#naming) should include the behavior being tested and expected outcomes. If we only care about the behavior, the scope is limited and each test is easy to understand. If we start to test real vendor names, what's the scope of it? Do we try to test all the available vendors? If not, why do we pick the selected vendors? 
That being said, if you feel strongly about testing with real vendor names, please add comments explaining the why these real vendors are selected and differences between each vendor.

Reply:
I have separated the real world tests and functionality tests, and given appropriate names and comments. Can you please review them once, and there's still any issue then we can continue the discussion?

Reply:
As for why we are using real vendor in testing is because this change is being done to capture the customer usage (which version/vendor/vendor.version the customer is using). If we make sure that we are able to capture real vendors information correctly, our usage data will be more accurate. As for the other functionality tests, I have changed the name and commented what is being tested. As for why these 3 real vendors are chosen. (1) graalvm - we support this and we have a specific dashboard for its usage, so for more accuracy. also because I changed its implementation (2) temurin - we officially support this.

Reply:
Thanks Deepankar! LGTM.

Reply:
I also want real world values used in unit tests. For future code readers, it works as a document of what kind of values the function deals with in real world.